src/transformers/generation/utils.py

```diff
@@ -1194,6 +1238,24 @@ def _validate_generated_length(self, generation_config, input_ids_length, has_de
                     UserWarning,
                 )
 
+    def _sanitize_kwargs(self, generation_config: GenerationConfig, kwargs: dict):
+        """
+        Sanitize the kwargs for generation
+        """
+        if getattr(generation_config, "cache_implementation", "") == CacheImplementation.SINK:
+            if "past_key_values" not in kwargs:
+                kwargs["past_key_values"] = SinkCache(
+                    window_length=generation_config.sink_window_length,
+                    num_sink_tokens=generation_config.num_sink_tokens,
+                )
```

Comment:
why not pass the class and do the same init scheme with generation config passed and you take these from the generation config to allow custom classes?